plugins/source/aws/client/client.go

```diff
@@ -115,7 +116,15 @@ func (c *Client) ID() string {
 	return strings.TrimRight(strings.Join(idStrings, ":"), ":")
 }
 
-func (c *Client) Services() *Services {
+func (c *Client) Services(service_names ...string) *Services {
+	for _, service := range service_names {
+		svcConfig := funk.Get(c.ServicesManager.ServicesByPartitionAccount(c.Partition, c.AccountID), strings.ToUpper(service[:1])+service[1:])
+		if svcConfig == nil {
+			c.accountMutex[c.AccountID].Lock()
+			c.ServicesManager.ServicesByPartitionAccount(c.Partition, c.AccountID).InitService(c.AWSConfig, service)
+			c.accountMutex[c.AccountID].Unlock()
```

Comment:
@hermanschaaf - The linter was telling me not to have a `defer c.accountMutex[c.AccountID].Unlock()` in a for loop. Now that `InitService` can panic we need a way to be able to unlock the mutex in that case. Any thoughts on how to structure the code other than ignoring the linting rule?

Reply:
Do we really need to unlock the mutex if the code panics? I would have thought all the goroutines get stopped anyway

Reply:
But alternatively you can move just this bit of code:

```
			c.accountMutex[c.AccountID].Lock()
			c.ServicesManager.ServicesByPartitionAccount(c.Partition, c.AccountID).InitService(c.AWSConfig, service)
			c.accountMutex[c.AccountID].Unlock()
```

into its own function like:

```
func (c *Client) update(...) {
			c.accountMutex[c.AccountID].Lock()
			defer c.accountMutex[c.AccountID].Unlock()
			c.ServicesManager.ServicesByPartitionAccount(c.Partition, c.AccountID).InitService(c.AWSConfig, service)
}
```

Reply:
This gets executed as part of a table resolver, so the go routine will get caught and I imagine the mutex would never be released

Reply:
Maybe I am wrong though?

Reply:
Oh I see yeah, we catch the panics from table resolvers. Ok well my second suggestion should work in that case